File: frontend/src/services/device.ts
import axios from 'axios';

// 类型定义
export type Device = {
  id: number;
  name: string;
  serial: string;
  model?: string;
  os_version?: string;
  status: 'online' | 'offline' | 'unauthorized' | 'unknown';
  connection_type: 'USB' | 'TCP/IP';
  ip_port?: string;
  capabilities?: string[];
  created_at: string;
  updated_at: string;
};

export type DeviceCreate = {
  name: string;
  serial: string;
  model?: string;
  os_version?: string;
  connection_type: 'USB' | 'TCP/IP';
  ip_port?: string;
  capabilities?: string[];
};

const api = axios.create({
  baseURL: '/api/v1',
  timeout: 30000,
});

// 获取设备列表
export const getDevices = (params?: {
  page?: number;
  page_size?: number;
  status?: string;
}) => {
  return api.get('/devices', { params });
};

// 获取设备详情
export const getDevice = (id: number) => {
  return api.get(`/devices/${id}`);
};

// 创建设备
export const createDevice = (data: DeviceCreate) => {
  return api.post('/devices', data);
};

// 更新设备
export const updateDevice = (id: number, data: Partial<DeviceCreate>) => {
  return api.put(`/devices/${id}`, data);
};

// 删除设备
export const deleteDevice = (id: number) => {
  return api.delete(`/devices/${id}`);
};

// 刷新设备状态
export const refreshDevices = () => {
  return api.post('/devices/refresh');
};

// 测试设备连接
export const testDeviceConnection = (serial: string) => {
  return api.post(`/devices/${serial}/test`);
};

// 获取设备DOM
export const getDeviceDom = (serial: string) => {
  return api.get(`/devices/${serial}/dom`);
};

File: backend/app/schemas/device.py
"""
Device schemas
"""
from typing import Optional
from pydantic import BaseModel, Field
from datetime import datetime


class DeviceCapabilitiesSchema(BaseModel):
    """Device capabilities"""
    biometric: bool = Field(False, description="Biometric support")
    nfc: bool = Field(False, description="NFC support")
    camera: bool = Field(False, description="Camera support")


class DeviceBase(BaseModel):
    """Base device fields"""
    name: str = Field(..., min_length=1, max_length=100, description="Device name")
    serial: str = Field(..., min_length=1, max_length=100, description="Device serial number")
    model: Optional[str] = Field(None, max_length=100, description="Device model")
    os_version: Optional[str] = Field(None, max_length=50, description="OS version")
    connection_type: str = Field("usb", pattern="^(usb|tcp)$", description="Connection type")
    ip_port: Optional[str] = Field(None, max_length=50, description="IP:Port for TCP")
    capabilities: Optional[DeviceCapabilitiesSchema] = Field(None, description="Device capabilities")


class DeviceCreate(DeviceBase):
    """Device creation schema"""
    pass


class DeviceUpdate(BaseModel):
    """Device update schema"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    model: Optional[str] = Field(None, max_length=100)
    os_version: Optional[str] = Field(None, max_length=50)
    connection_type: Optional[str] = Field(None, pattern="^(usb|tcp)$")
    ip_port: Optional[str] = Field(None, max_length=50)
    capabilities: Optional[DeviceCapabilitiesSchema] = None


class DeviceResponse(DeviceBase):
    """Device response schema"""
    id: int
    adb_status: str = Field(..., description="ADB status: online, offline, unauthorized")
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
